### PR TITLE
refactor(Project Authoring): Extract move node location selection feature to component

### LINKED ISSUE #1393: refactor(Project Authoring): Extract move node location selection to new component
In the project authoring view, asking the author to select a new location for a moved node(s) is done in the same component (ProjectAuthoringComponent). This should be done in a separate component to reduce responsibility of this component.

To limit scope, handle extracting copied component in another task.

Look at ChooseComponentLocationComponent for reference.

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Notes
- Please style and clean code/template as you see fit

## Changes
- Extract mode node location selection feature from ProjectAuthoring to ChooseMoveNodeLocationComponent
- Clean up related code

## Test
- Move node works as before:
   - one node at a time
   - multiple nodes at a time
   - steps
   - groups
   - to/from active and inactive sections 

Closes #1393

### CODE REVIEW

File: src/assets/wise5/authoringTool/choose-move-node-location/choose-move-node-location.component.html [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            *ngTemplateOutlet="moveAfterButton; context: { nodeId: nodeId }"
+          ></ng-container>
+        </ng-container>
+      </div>
+    </div>
+  </div>
+</div>
+<div>
+  <div fxLayout="row" fxLayoutAlign="start center" fxLayoutGap="10px">
+    <h6 class="underline">Unused Lessons</h6>
```
Comment (geoffreykwan) [inline src/assets/wise5/authoringTool/choose-move-node-location/choose-move-node-location.component.html]: Add i18n to the "Unused Lessons" element.

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
Functionality works well.

I'm wondering if instead of using ng-templates it would be better to create components. It would simplify the html and the components could be reused in other templates.

This looks like code duplication. It seems like Code Climate doesn't check html files?

https://github.com/WISE-Community/WISE-Client/blob/0497dd7d76ebd7ac9e2ee68391cc024e8e8d312f/src/assets/wise5/authoringTool/choose-move-node-location/choose-move-node-location.component.html#L118-L125

https://github.com/WISE-Community/WISE-Client/blob/0497dd7d76ebd7ac9e2ee68391cc024e8e8d312f/src/assets/wise5/authoringTool/choose-move-node-location/choose-move-node-location.component.html#L142-L149

File: src/assets/wise5/authoringTool/choose-move-node-location/choose-move-node-location.component.html (hunk 1 above)
Comment (hirokiterashima) [reply src/assets/wise5/authoringTool/choose-move-node-location/choose-move-node-location.component.html]: Done. We'll hold off on making components until we extract ChooseCopyNodeLocationComponent from ProjectAuthoring. And yes, it seems like CodeClimate does not check html atm. Maybe we told it to ignore html files?

**Review by breity — APPROVED** [review]
Looks good. I cleaned up some styles and removed unused classes.

I also updated margin styles to be friendly to RTL languages.

**Comment by hirokiterashima** [comment]
Thanks @breity. Good to remove unused styles. Template files look cleaner now.

@geoffreykwan can you take a final look?

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
When you move a step or a lesson and are in the view to choose the location, the unused lessons and unused steps are not indented correctly.

**Comment by hirokiterashima** [comment]
Fixed. PTAL.

**Review by geoffreykwan — APPROVED** [review]
Looks good.

**Pull request merged**